backend/pkg/proto/service.go

```diff
@@ -251,12 +282,35 @@ func (s *Service) getMessageDescriptorFromConfluentMessage(wrapper *confluentEnv
 	return msgType, wrapper.ProtoPayload, nil
 }
 
+func (s *Service) getMatchingMapping(topicName string) (mapping config.ProtoTopicMapping, err error) {
+	mapping, strictExists := s.strictMappingsByTopic[topicName]
+	if strictExists {
+		return mapping, nil
+	}
+
+	var match bool
+	for _, rMapping := range s.regexMappingsByTopic {
+		match = rMapping.r.MatchString(topicName)
```

Comment:
This is a performance degradation for a non-matched topic.
Each time for non-matched topics, the algorithm iterate and match all known regex.

The maximum number of searches has to be estimated as total unique topic names O(N).

I suggest saving the `nil` result to the strict mapping `s.strictMappingsByTopic[topicName]` and checking on the `nil` output at the beginning of this function to avoid another loop searching by regex.